airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/GenerateInputActivityImpl.java

```diff
@@ -90,25 +90,18 @@ public GeneratedJobInput getSyncWorkflowInput(final SyncInput input) {
 
       final JobRunConfig jobRunConfig = TemporalWorkflowUtils.createJobRunConfig(jobId, attempt);
 
-      List<StandardDestinationDefinition> destinationDefinitionList = configRepository.listStandardDestinationDefinitions(true);
-      Optional<StandardDestinationDefinition> optionalDestinationDefinition = destinationDefinitionList.stream()
-          .filter(destinationDefinition -> config.getDestinationDockerImage()
-              .equalsIgnoreCase(
-                  DockerUtils.getTaggedImageName(destinationDefinition.getDockerRepository(), destinationDefinition.getDockerImageTag())))
-          .findFirst();
-      final String destinationNormalizationDockerImage = optionalDestinationDefinition
-          .filter(standardDestinationDefinition -> Objects.nonNull(standardDestinationDefinition.getNormalizationConfig()))
-          .map(standardDestinationDefinition -> String.format("%s:%s",
-              standardDestinationDefinition.getNormalizationConfig().getNormalizationRepository(),
-              standardDestinationDefinition.getNormalizationConfig().getNormalizationTag()))
-          .orElse(null);
-      final boolean supportstDbt = optionalDestinationDefinition.isPresent() && Objects.nonNull(optionalDestinationDefinition.get().getSupportsDbt())
-          ? optionalDestinationDefinition.get().getSupportsDbt()
-          : false;
-      final String normalizationIntegrationType = optionalDestinationDefinition
-          .filter(standardDestinationDefinition -> Objects.nonNull(standardDestinationDefinition.getNormalizationConfig()))
-          .map(standardDestinationDefinition -> standardDestinationDefinition.getNormalizationConfig().getNormalizationIntegrationType())
-          .orElse(null);
+      final UUID connectionId = UUID.fromString(job.getScope());
+      final StandardSync standardSync = configRepository.getStandardSync(connectionId);
+
+      final StandardDestinationDefinition destinationDefinition =
+          configRepository.getDestinationDefinitionFromDestination(standardSync.getDestinationId());
+      final String destinationNormalizationDockerImage = destinationDefinition.getNormalizationConfig() != null
+          ? DockerUtils.getTaggedImageName(destinationDefinition.getNormalizationConfig().getNormalizationRepository(),
+              destinationDefinition.getNormalizationConfig().getNormalizationTag())
+          : null;
+      final String normalizationIntegrationType =
+          destinationDefinition.getNormalizationConfig() != null ? destinationDefinition.getNormalizationConfig().getNormalizationIntegrationType()
+              : null;
```

Comment:
While I was in here, I refactored how we were looking up the destination definition from the sync. We need to get the StandardSync anyway to retrieve the sourceId and destinationId for config updating now, so we can simplify to get the definition from the destinationId directly. 

This used to be looking up all destination definitions and filtering based on docker image + tag combo, which also seemed a bit error prone if we decide to move into a world where multiple definitions could have the same docker image (which I think _will_ be the case)

The functionality should be equivalent.